### PR TITLE
Feature/armor progression

### PR DESCRIPTION
**Pull request by GildedToxin**
Diamond -> Netherite -> Dark metal -> Etherium/Enderium/Adamantite-> Ignitium

Heavy/Adamantite - less move speed, more knock back resistance, more health, more armor, more toughness, Less attack damage

Normal/Enderium - Armor up, Toughness up

Light/Etherium - Less health, less armor, less tough, More attack damage

Ignitium - more armor, more tougness, more health, more attack

### CODE REVIEW

File: overrides/scripts/modstats/armor_progression/neptunium_armor.zs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import crafttweaker.api.entity.attribute.AttributeOperation;
+
+<item:aquaculture:neptunium_helmet>.anyDamage().addGlobalAttributeModifier(<attribute:minecraft:generic.armor_toughness>, "Nerf_Dark_Metal_Helmet_Toughness", 0.5, AttributeOperation.ADDITION, [<constant:minecraft:equipmentslot:head>]);
+<item:aquaculture:neptunium_chestplate>.anyDamage().addGlobalAttributeModifier(<attribute:minecraft:generic.armor_toughness>, "Nerf_Dark_Metal_Chest_Toughness", 0.5, AttributeOperation.ADDITION, [<constant:minecraft:equipmentslot:chest>]);
+<item:aquaculture:neptunium_leggings>.anyDamage().addGlobalAttributeModifier(<attribute:minecraft:generic.armor_toughness>, "Nerf_Dark_Metal_Leggings_Toughness", 0.5, AttributeOperation.ADDITION, [<constant:minecraft:equipmentslot:legs>]);
+<item:aquaculture:neptunium_boots>.anyDamage().addGlobalAttributeModifier(<attribute:minecraft:generic.armor_toughness>, "Nerf_Dark_Metal_Boots_Toughness", 0.5, AttributeOperation.ADDITION, [<constant:minecraft:equipmentslot:feet>]);
```
Comment (Jonah-Hansen) [inline overrides/scripts/modstats/armor_progression/neptunium_armor.zs:6]: attribute modifier names are Dark Metal here should be neptunium

**Review by Jonah-Hansen — CHANGES_REQUESTED** [review]
- theres an empty `armor_upgrades.zs` file in `/upgrading` I assume can be deleted.

- `amethyst_armor.zs` is all by itself in `/modstats` not sure if it should be moved into `/armor_progression` or just stay 

- should probably remove `test.zs` in `/armour_progression`


is it possible to modify the "when on body/legs/feet/head" attributes text so that similar stat changes are actually combined? ei. adamantite chestplate: "+2.5 armor, +8 armor" to just say +10.5 armor? 

or if not can we add some sort of other description text to explain the differences between the endgame pieces? I know we'll probably explain it in the quest book but its nice to hover over them in jei and see the differences there.

**Review by Jonah-Hansen — CHANGES_REQUESTED** [review]
adding uuids fixed the issue of combining stat changes! please take a look at `/utils/AttributeIds` and add the remaining uuids. then we can use `AttributeIds.boots` etc in place of the uuid string. this solves the bad practice of having "magic strings" in our code!

please apply the uuid method to all the adjusted armors.

**Comment by GildedToxin** [comment]
I'm like half way through em
didnt even realize that it was the same for all of them 0_o
this makes my life easier ig
but I question the time I spent
thank you none the less

**Comment by GildedToxin** [comment]
I did it for all tools stats while I was at it

**Review by Jonah-Hansen — APPROVED** [review]
tested looks great! 

full adamantite is VERY slow.. I know people will generally mix armor pieces but if someone did want to be a full tank before 16 eyes, its almost unplayable with the speed reduction. 
would love to hear other opinions!

**Comment by GildedToxin** [comment]
Very much agreed with the adamantite being slow - I wasn't really sure how slow to make it
I intended players to use other means to balance out the speed reduction, such as speed potions, beacon, and running shoes (or mix and match the armor like you said) - also if they are in the end, they have elyra which will be there main mode of movement
I could also give the armor more health as way to balance out the heavy speed loss
I've played 4 hours, so im not really sure how much health is `worth it` to players
I could also always just cut down on the speed and call it a day

**Comment by Jonah-Hansen** [comment]
I vote just cut down on speed reduction by 25-50% of what it currently is. by late game everyone has speed boost items anyways so it just feels like a pretty heavy nerf. I think 4 extra hearts with a full set is pretty good.

**Comment by GildedToxin** [comment]
rn it removes 80% move speed, ill knock down to 60%/50%
when you says 4 hearts do you mean for full hearts? or 4 halfs which make 2 full hearts (what the armor has rn)?

**Comment by Jonah-Hansen** [comment]
tbh I thought it was 4 full. tough to decide. 2 full doesn't seem like much when considering ghast eye but also 4 full is probably a lot. maybe just leave it as as and we can reevaluate after playing with it for real

**Comment by RichPrune** [comment]
gaming

**Comment by GildedToxin** [comment]
gaming but no approval 😞

**Pull request merged**